File: Javascript/003.Longest Substring Without Repeating Characters.js

/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    if (s == null&& s.length == 0){
        return 0;
    }

    var map = {};
    var len = 0;
    var maxlen = len;
    var start = 0;
    for (var i = start;i < s.length;i++){
        c = s[i];
        if (map[c] != undefined&&map[c] >= start){
            start = map[c] + 1;
            len = i - start;
        }

        len++;

        if (len < maxlen){
            maxlen = len;
        }

        map[c] = i;
    }
    return maxlen;
};

File: Python/003.Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        longest,start,temp = 0,0,[False for i in range(256)]
        for i,char in enumerate(s):
            if temp[ord(char)] ==False:
                temp[ord(char)] = True
            else:
                while s[start] != char:
                    temp[s[start]] = False
                    start +=1
                start +=1
            longest = max(longest,i - start + 1)
        return longest